### ISSUE TITLE
The fitting nerf is at a standstill

### ISSUE THREAD

**Issue by QinglinZhan** [gzhang292/MultiGO#2]
I'm very happy with your work and thank you very much for the additions to the code. When I was running the last version of your code, I had a long stagnation problem at step 7 in the process of reasoning about the various materials available, and I don't know how to solve it, so I hope you can give me some advice, thank you very much.

`  warnings.warn("xFormers is available (Attention)")
Warp 1.7.2 initialized:
   CUDA Toolkit 12.8, Driver 12.4
   Devices:
     "cpu"      : "x86_64"
     "cuda:0"   : "NVIDIA GeForce RTX 3090" (24 GiB, sm_86, mempool enabled)
   Kernel cache:
     /home/cver2/.cache/warp/1.7.2
Step 7: Convert glb...
Running command: python convert.py big --test_path .././ood_1_recontructed_gaussians/ood/gaussian_pred_infer_0.ply
Number of points at loading :  22820
[INFO] fitting nerf...
  0%|                                                                                                                                                                         | 0/512 [00:00<?, ?it/s]`

**Comment by Jackiemin233** [comment]
This code is derived from the GS2Mesh conversion part of the LGM method. To solve this problem, could you please debug this code and tell me where it stuck so that I can help you more effectively.

**Comment by songget** [comment]
> This code is derived from the GS2Mesh conversion part of the LGM method. To solve this problem, could you please debug this code and tell me where it stuck so that I can help you more effectively.

Thank you very much for your willingness to help me answer this question, in the process of debugging the code, I found that the program is unresponsive to the loop iteration process in the def fit_nerf part of convert.py, and I wonder if you can run it successfully or help me answer the question.
`
    def fit_nerf(self, iters=512, resolution=256):

        self.opt.output_size = resolution

        optimizer = torch.optim.Adam([
            {'params': self.encoder_density.parameters(), 'lr': 1e-2},
            {'params': self.encoder.parameters(), 'lr': 1e-2},
            {'params': self.mlp_density.parameters(), 'lr': 1e-3},
            {'params': self.mlp.parameters(), 'lr': 1e-3},
        ])

        print(f"[INFO] fitting nerf...")
        pbar = tqdm.trange(iters)
        for i in pbar:

            ver = np.random.randint(-45, 45)
            hor = np.random.randint(-180, 180)
            rad = np.random.uniform(1.5, 3.0)
            
            pose = orbit_camera(ver, hor, rad)
            
            image_gt, alpha_gt = self.render_gs(pose, size=resolution)
            image_pred, alpha_pred = self.render_nerf(pose)

            # if i % 200 == 0:
            #     kiui.vis.plot_image(image_gt, alpha_gt, image_pred, alpha_pred)
            
            loss_mse = F.mse_loss(image_pred, image_gt) + 0.1 * F.mse_loss(alpha_pred, alpha_gt)
            loss = loss_mse #+ 0.1 * self.encoder_density.tv_loss() #+ 0.0001 * self.encoder_density.density_loss()

            loss.backward()
            self.encoder_density.grad_total_variation(1e-8)
        
            optimizer.step()
            optimizer.zero_grad()

            pbar.set_description(f"MSE = {loss_mse.item():.6f}")
        
        print(f"[INFO] finished fitting nerf!")
`

**Comment by Jackiemin233** [comment]
It would be better If you can provide which line of code it got stuck. Generally, it's caused by the incompatible versions between CUDA and PyTorch. You can try to check it with command 'nvcc -V' and reinstall the the PyTorch package.

**Comment by songget** [comment]
Thanks for the heads up, I checked my torch and CUDA and the versions of both match each other. However when I step-by-step debug the code and run to this section of code, it is in a long stuttering state with no output and no subsequent code running.

`with` torch.no_grad(): 
ray_indices, t_starts, t_ends = self.estimator.sampling( 
rays_o, 
rays_d, 
sigma_fn=sigma_fn, 
near_plane=0.01, 
far_plane=100, 
render_step_size=self.render_step_size, 
stratified=self.training, cone_angle=0, 
)
`
